app/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/DrawAreaTaskFragment.kt

```diff
@@ -104,6 +112,30 @@ class DrawAreaTaskFragment @Inject constructor() : AbstractTaskFragment<DrawArea
     if (isVisible && !viewModel.instructionsDialogShown) {
       showInstructionsDialog()
     }
+    viewLifecycleOwner.lifecycleScope.launch {
+      viewModel.showSelfIntersectionDialog.collect {
+        renderComposableDialog { SelfIntersectionDialog() }
+      }
+    }
+  }
+
+  @Composable
+  fun SelfIntersectionDialog() {
+    val showDialog = remember { mutableStateOf(true) }
+    if (showDialog.value) {
+      AlertDialog(
+        onDismissRequest = {},
+        title = {
+          Text(text = stringResource(R.string.polygon_vertex_add_dialog_title), fontSize = 18.sp)
+        },
+        text = { Text(text = stringResource(R.string.polygon_vertex_add_dialog_message)) },
+        confirmButton = {
+          OutlinedButton(onClick = { showDialog.value = false }) {
+            Text(text = stringResource(R.string.polygon_vertex_add_dialog_positive_button))
+          }
+        },
+      )
+    }
```

Comment:
Can we reuse the compose component `ConfirmationDialog` instead of building a custom one?

Reply:
need to make `dismissButtonText` optionale, is it ok or will it affect any place?